File: engine/model/src/main/scala/tools/packet/foreigneffect/ShowBerserk.scala
package tools.packet.foreigneffect

import net.opcodes.SendOpcode
import tools.packet.PacketInput

class ShowBerserk(private var _characterId: Int, private var _skillLevel: Int, private var _berserk: Boolean) extends PacketInput {
  def characterId: Int = _characterId

  def skillLevel: Int = _skillLevel

  def berserk: Boolean = _berserk

  override def opcode(): SendOpcode = SendOpcode.SHOW_FOREIGN_EFFECT
}

File: engine/engine-model/src/main/java/tools/packet/foreigneffect/ShowBerserk.java
package tools.packet.foreigneffect;

import net.opcodes.SendOpcode;
import tools.packet.PacketInput;

public record ShowBerserk(Integer characterId, Integer skillLevel, Boolean berserk) implements PacketInput {
   @Override
   public SendOpcode opcode() {
      return SendOpcode.SHOW_FOREIGN_EFFECT;
   }
}